src/components/app/exclusioncones/Form/ExclusionForm.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { useEffect, useState } from "react";
+import { useForm, FormProvider } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { Button } from "@/components/ui/button";
+import { Separator } from "@/components/ui/separator";
+import axios from "axios";
+import { toast } from "@/hooks/use-toast";
+import {
+  ExclusionFormSchema,
+  ExclusionFormData,
+} from "@/components/app/exclusioncones/Form/ExclusionFormSchema";
+import FormFieldWrapper from "@/components/ui/wrapper/formfieldwrapper";
+import FormCombobox, { Option } from "@/components/ui/wrapper/comboboxwrapper";
+
+const ExclusionForm: React.FC = () => {
+  const form = useForm<ExclusionFormData>({
+    resolver: zodResolver(ExclusionFormSchema),
+    defaultValues: {
+      mission: "",
+      angle_limit: 0,
+      interfering_satellite: "",
+      satellite_id: "",
+      gs_id: 0,
+    },
+  });
+
+  // Option states for Satellite and Ground Station selection.
+  const [satelliteOptions, setSatelliteOptions] = useState<Option[]>([]);
+  const [groundStationOptions, setGroundStationOptions] = useState<Option[]>([]);
+
+  useEffect(() => {
+    async function fetchSatellites() {
+      try {
+        const response = await axios.get(
+          process.env.NEXT_PUBLIC_API_URL + "/api/v1/satellites"
+        );
+        // Map the API response so that:
+        // - value is set to the satellite's "id"
+        // - label is set to the satellite's "name"
+        const options = response.data.map((sat: any) => ({
+          value: sat.id,
+          label: sat.name,
```

Comment:
The issue here is that there could potentially be multiple satellites with the same name (since we are not enforcing name uniqueness) and you would end up with something shown in the SS. Also having UUIDs visible would be helpful, especially since we can use the search feature.


My suggestion would be to show both name and ID.
```suggestion
          label: `${sat.name} (${sat.id})`,
```

Now the issue is the text will overflow. To address that for now you can add 
`overflow-hidden text-ellipsis whitespace-nowrap` to the `className` of the `FormBox`

![Screenshot 2025-03-21 at 10 58 53 PM](https://github.com/user-attachments/assets/0ae0e604-f621-4f54-9da7-7e987fe32676)